torch/_functorch/aot_autograd.py

```diff
@@ -1050,9 +1050,52 @@ class AOTConfig:
     num_params_buffers: int
     aot_id: int
 
+def aot_dispatch_base_traced_fn(flat_fn, meta, synthetic_base_info, *flat_args_with_views_handled):
```

Comment:
This looks suspiciously duped and not deduplicated from the other code

Reply:
forgot to leave a comment but yes: the function that we want to trace for `aot_dispatch_autograd` is a superset of what we want to trace for `aot_dispatch_base`, so we can definitely share code.

Originally I felt that `create_joint_forward_backward_functionalized` was already complex enough and trying to factor it to handle the non-autograd case would make it unwieldy, but I think I've factored it in a way that makes it clearer (fresh commit coming in a bit). But lmk what you think

Reply:
To be clear, the original `create_joint_forward_backward_functionalized` is complicated. But I have been pushing for you to figure out how to factor it into smaller composable pieces, to make it less complicated. From adding inference support here, it seems clear that there is a case for making this composable is useful in any case.